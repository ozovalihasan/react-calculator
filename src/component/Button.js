import React from 'react';
import PropTypes from 'prop-types';
import '../index.scss';

const Button = ({ name, wide, color }) => {
  const buttonPropStyle = {
    width: `${wide ? '50%' : '25%'}`,
    backgroundColor: color,
  };

  return (
    <div className="button" style={buttonPropStyle}>
      {name}
    </div>
  );
};

Button.propTypes = {
  name: PropTypes.string,
  wide: PropTypes.bool,
  color: PropTypes.string,
};

Button.defaultProps = {
  name: '',
  wide: false,
  color: 'orange',
};

export default Button;
