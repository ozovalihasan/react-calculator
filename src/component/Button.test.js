import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { screen, fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Button from './Button';

let container = null;
let clickHandler;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  clickHandler = jest.fn();
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('creates a button with given click handler function, name, default width and color', () => {
  act(() => {
    render(<Button name="big_button" clickHandler={clickHandler} />, container);
  });
  expect(container.textContent).toBe('big_button');
  expect(screen.getByText(/big_button/).className).toBe('button main w-25');
  expect(screen.getByText(/big_button/).style['background-color']).toBe(
    'darkgray',
  );
  act(() => {
    fireEvent.click(screen.getByText(/big_button/));
  });
  expect(clickHandler).toHaveBeenCalledTimes(1);
});

it('creates a button with given click handler function, name, larger width and default background color', () => {
  act(() => {
    render(
      <Button name="big_button" wide clickHandler={clickHandler} />,
      container,
    );
  });
  expect(screen.getByText(/big_button/).className).toBe('button main w-50');
});

it('creates a button with given click handler function, name, background color and default width', () => {
  act(() => {
    render(
      <Button name="big_button" color="orange" clickHandler={clickHandler} />,
      container,
    );
  });
  expect(screen.getByText(/big_button/).style['background-color']).toBe(
    'orange',
  );
});
