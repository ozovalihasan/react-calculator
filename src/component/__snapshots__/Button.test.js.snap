// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="button main w-25"
  onClick={[MockFunction]}
  style={
    Object {
      "backgroundColor": "darkgray",
    }
  }
  type="button"
>
  big_button
</button>
`;
